packages/gatsby-core-utils/src/fetch-remote-file.ts

```diff
@@ -159,15 +161,35 @@ const requestRemoteNode = (
     })
   })
 
+const fetchCache = new Map()
+
 export async function fetchRemoteFile({
+  skipCache,
+  ...args
+}: IFetchRemoteFileOptions): Promise<string> {
+  // If we are already fetching the file, return the unresolved promise
+  const inFlight = fetchCache.get(args.url)
+  if (inFlight && !skipCache) {
+    return inFlight
+  }
+
+  // Create file fetch promise and store it into cache
+  const fetchPromise = fetchFile(args)
+  fetchCache.set(args.url, fetchPromise)
+
+  return fetchPromise
```

Comment:
We should remove it from the fetch cache on error so you can drop the skipCache :)

```suggestion
  return fetchPromise.catch(err => {
    fetchCache.remove(args.url);
    
    throw err;
  })
```

Reply:
Great! Done!